ansible_collections/arista/avd/docs/getting-started/avd-aap.md

```diff
@@ -0,0 +1,403 @@
+
+### Ansible Builder
+
+Ansible Builder is a tool developed by the Ansible team to aid in creating EEs. Like any tool, there are a few requirements to get started with Ansible Builder.
+
+- A development machine with Python installed.
+- Ansible builder installed with `pip install ansible-builder`.
+- [Docker](https://docs.docker.com/engine/install/) or [Podman](https://podman.io/docs/installation) installed on the development machine.
+
+You can place the Ansible builder dependencies with your current project or leverage a separate project entirely for your EE builds. Please note that this example was created with version 4.7.1 of the AVD collection. Dependencies may change between versions.
+
+```shell
+❯ tree -I venv/
+.
+├── execution-environment.yml
+├── requirements.txt
+└── requirements.yml
+```
+
+=== "execution-environment.yml"
+
+    ```yaml
+    ---
+    version: 3 #(1)
+
+    images: #(2)
+      base_image:
+        name: quay.io/centos/centos:stream8
+
+    dependencies: #(3)
+      python_interpreter:
+        package_system: python311
+        python_path: /usr/bin/python3.11
+
+      ansible_core: #(4)
+        package_pip: ansible-core<2.17.0
+
+      ansible_runner: #(5)
+        package_pip: ansible-runner
+
+      galaxy: requirements.yml #(6)
+
+      python: requirements.txt #(7)
+
+    ```
+
+    1. We are leveraging Ansible Builder 3.x, which requires version 3 of the definition file. If you leverage an older version of Ansible Builder, you may need to use `version: 1`.
+    2. You may use any base container image. Please see the official documentation for more examples.
+    3. Specifying what version of Python we would like installed during the container build.
+    4. The version of ansible-core you need will greatly depend on your specific workflows.
+    5. The `ansible-runner` package is required when building an EE.
+    6. This is a pointer to any additional collections we want installed on our container.
+    7. This is a pointer to any additional Python packages we require for our workflow.
+
+=== "requirements.yml"
+
+    ```yaml
+    ---
+    collections: #(1)
+      - name: arista.avd
+        version: 4.7.1
+
+    ```
+
+    1. Installing the `arista.avd` collection for this workflow will ensure that any other required collections are also installed.
+
+=== "requirements.txt"
+
+    ```text
+    netaddr>=0.7.19
+    Jinja2>=3.0.0
+    treelib>=1.5.5
+    cvprac>=1.3.1
+    jsonschema>=4.5.1,<4.18
+    requests>=2.27.0
+    PyYAML>=6.0.0
+    md-toc>=7.1.0
+    deepmerge>=1.1.0
+    cryptography>=38.0.4
+    # No anta requirement until the eos_validate_state integration is out of preview.
+    # anta>=1.0.0
+    aristaproto>=0.1.1
+
+    ```
```

Comment:
I'd update that as well as it's now very different.

Reply:
True but I do mention to the user to check the docs for the latest requirements pertaining to their chosen version and requirements may change between releases

Reply:
you should always just snip in the requirements file so it is always up to date.
```suggestion
=== "requirements.txt"

    ```text
    --8<-- "requirements.txt"
    ```
```

Reply:
Ill update this to only have the pyavd requirement. This would not work previously if the guide was written for say 4.6 and we updated the requirements file that would not work with 4.6 installs